backend/app.py

```diff
@@ -59,6 +75,108 @@ def post_item():
     return json.dumps({'success': True, 'error': None})
 
 
+@app.route('/price', methods=['POST'])
+def add_price():
+    '''
+        Body: {"upc", "price", "user", "store", "lat", "long"}
+        Response:
+            - {"success": true or false},
+            - {"error": error description}
+    '''
+    data = request.get_json(force=True)
+
+    required_fields = ['upc', 'price', 'user', 'store', 'lat', 'long']
+    if not validation.has_required(data, required_fields):
+        return json.dumps({'success': False, 'error': Error.MISSING_FIELDS.value})
+
+    item = model.Item.objects(upc=data['upc']).first()
+    if item is None:
+        return json.dumps({'success': False, 'error': Error.ITEM_DNE.value})
+
+    new_price = model.Price(
+        user=data['user'],
+        upvotes=[],
+        downvotes=[],
+        price=data['price'],
+        date=request.date
+    )
+
+    loc = {'lat': data['lat'], 'long': data['long']}
+    store = item.stores.filter(name=data['store'], location=loc).first()
+    if store is not None:
+        store.prices.append(new_price)
+    else:
+        new_store = model.Store(
+            name=data['store'],
+            location=loc,
+            prices=[new_price]
+        )
+        item.stores.append(new_store)
+
+    try:
+        item.save()
+    except Exception as e:
+        return json.dumps({'success': False, 'error': str(e)})
+
+    return json.dumps({'success': True, 'error': None})
+
+
+@app.route('/vote', methods=['POST'])
+def vote():
+    '''
+        Body: {"upc", "user", "store", "lat", "long", "dir"}
+        Response:
+            - {"success": true or false},
+            - {"error": error description}
+    '''
+    data = request.get_json(force=True)
+
+    required_fields = ['upc', 'user', 'store', 'lat', 'long', 'dir']
+    if not validation.has_required(data, required_fields):
+        return json.dumps({'success': False, 'error': Error.MISSING_FIELDS.value})
+    if not validation.is_valid_dir(data['dir']):
+        return json.dumps({'success': False, 'error': Error.INVALID_DIR.value})
+
+    item = model.Item.objects(upc=data['upc']).first()
+    if item is None:
+        return json.dumps({'success': False, 'error': Error.ITEM_DNE.value})
+
+    loc = {'lat': data['lat'], 'long': data['long']}
+    store = item.stores.filter(name=data['store'], location=loc).first()
+    if store is None:
+        return json.dumps({'success': False, 'error': Error.STORE_DNE.value})
+    else:
+        price = store.prices[-1]
+        if data['dir'] == 1:
```

Comment:
Consider to abstract 1 and -1 as some Enum values, too? (i.e. VOTE_DIRECTION_UP, you can figure out better name than I do)